test/e2e/webdriver/driver.js

```diff
@@ -862,6 +862,42 @@ class Driver {
     return await this.windowHandles.getAllWindowHandles();
   }
 
+  /**
+   * Function that aims to simulate a click action on a specified web element
+   * within a web page and waits for the current window to close.
+   *
+   * @param {string | object} rawLocator - Element locator
+   * @param {number} [retries] - The number of times to retry the click action if it fails
+   * @returns {Promise<void>} promise that resolves to the WebElement
+   */
+  async clickElementAndWaitForWindowToClose(rawLocator, retries = 3) {
+    const handle = await this.driver.getWindowHandle();
+    await this.clickElement(rawLocator, retries);
+    await this.waitForWindowToClose(handle);
+  }
+
+  /**
+   * Waits for the specified window handle to close before returning.
+   *
+   * @param {string} handle - The handle of the window or tab we'll wait for.
+   */
+  async waitForWindowToClose(handle) {
+    let start = Date.now();
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const handles = await this.getAllWindowHandles();
+      if (!handles.includes(handle)) {
+        break;
+      }
+      if (start && Date.now() - start > 15000) {
+        console.info(
+          'The window did not close after 15 seconds, are you sure you closed it?',
+        );
+        start = null; // only log once
```

Comment:
If the window never closes, this will run until the test times out (usually 80 seconds total).  It also fails with the error
`Error: Timeout of 80000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.`

I don't think that's what we want, I think we want to throw a useful error here.

Reply:
There is no error in this method to throw, the test timing out on its own already covers the time out case. This section here is just to help a develop that called this method without understanding that they are responsible for closing the tab.

Reply:
Maybe we could allow passing a(n optional?) `timeout` value here?